src/nfa/nfa.cc

```diff
@@ -472,6 +503,25 @@ State Delta::find_max_state() {
     return max;
 }
 
+std::vector<Post> Delta::transform(const std::function<State(State)>& lambda) const {
```

Comment:
Is there a reason for why this is not in place? Or should it be?

Reply:
Yes I have made it deliberately const. It avoids copying of Delta inside concatenation. The second thing is that after in-place modification Delta could by inconsistent.

Reply:
Ok, I get there is some reason, but I still don't get what it is. Is it perhaps possible to write things in  even more idiot friendly way, at least when there is time?

Reply:
This is what was designed in C++ to create a simple interface for passing functions around. See [SO answer](https://stackoverflow.com/a/20353798) explaining what `std::function` is and how to use it.